boards/arm/lpcxpresso54114/pinmux.c

```diff
@@ -90,6 +89,27 @@ static int lpcxpresso_54114_pinmux_init(struct device *dev)
 			IOCON_PIO_OPENDRAIN_DI
 			);
 	pinmux_pin_set(port0,  4, port0_pin4_config);
+
+#ifdef DT_INST_0_TEST_GPIO_BASIC_API
```

Comment:
Agree that test specific stuff should go inside the test's code, but i find it messy.
I find using board's pinmux to be a cleaner approach whether it's an application, sample or test.
I also mentioned this in https://github.com/zephyrproject-rtos/zephyr/pull/19247#discussion_r332057888

Reply:
> I find using board's pinmux to be a cleaner approach

I agree, but at the moment all other drivers placed the pinmux configuration directly in the testcase. We know that what we're currently doing is not a clean approach but there was no time to do it properly. This will be fixed. It will be easier to work on the fix if all the drivers use the same approach for now. The same applies to `boards/arm/lpcxpresso55s69/pinmux.c`.

Reply:
Actually only frdm_k64f board has its pinmux configuration done in the testcase at the moment. Other PRs are not yet merged. Doing pinmux configuration directly in the board file makes sense but it really shouldn't be testcase specific. If we follow this road the board files will quickly become unmaintainable. Unfortunately, the approach to running tests on hardware is not fully worked out at the moment.